src/lib/io/include/sol-uart.h

```diff
@@ -224,21 +239,35 @@ struct sol_uart *sol_uart_open(const char *port_name, const struct sol_uart_conf
 void sol_uart_close(struct sol_uart *uart);
 
 /**
- * @brief Perform a UART asynchronous transmission.
+ * @brief Perform an UART asynchronous transmission.
+ *
+ * @param uart The UART bus handle
+ * @param blob The blob to be written
+ * @return 0 on success negative errno on error
+ *
+ */
+int sol_uart_write(struct sol_uart *uart, struct sol_blob *blob);
+
+/**
+ * @brief Perform an UART synchronous read
+ *
+ * @param uart The UART bus handle
+ * @param rx Where to store the UART data
+ * @param length number of bytes to read
```

Comment:
docs makes no sense, and document what is expected from buf (it must be initialized and all). Also document this "read to buffer" in a general section since this is repeating in many places.